core/schema/modulesource_test.go

```diff
@@ -53,16 +56,133 @@ func TestParsePublicRefString(t *testing.T) {
 				kind:           core.ModuleSourceKindGit,
 				repoRoot:       &vcs.RepoRoot{Root: "github.com/shykes/daggerverse.git", Repo: "https://github.com/shykes/daggerverse"},
 				repoRootSubdir: "../../",
+				scheme:         core.NoScheme,
+				sshusername:    "",
 			},
 		},
-		// gitlab
+		{
+			urlStr: "git+https://github.com/shykes/daggerverse/ci",
+			want: &parsedRefString{
+				modPath:        "github.com/shykes/daggerverse/ci",
+				kind:           core.ModuleSourceKindGit,
+				repoRoot:       &vcs.RepoRoot{Root: "github.com/shykes/daggerverse", Repo: "https://github.com/shykes/daggerverse"},
+				repoRootSubdir: "ci",
+				scheme:         core.SchemeGitHTTPS,
+				sshusername:    "",
+			},
+		},
+		{
+			urlStr: "git+http://github.com/shykes/daggerverse.git/ci",
+			want: &parsedRefString{
+				modPath:        "github.com/shykes/daggerverse.git/ci",
+				kind:           core.ModuleSourceKindGit,
+				repoRoot:       &vcs.RepoRoot{Root: "github.com/shykes/daggerverse.git", Repo: "https://github.com/shykes/daggerverse"},
+				repoRootSubdir: "ci",
+				scheme:         core.SchemeGitHTTP,
+				sshusername:    "",
+			},
+		},
+		{
+			urlStr: "git+ssh://github.com/shykes/daggerverse.git/ci",
+			want: &parsedRefString{
+				modPath:        "github.com/shykes/daggerverse.git/ci",
+				kind:           core.ModuleSourceKindGit,
+				repoRoot:       &vcs.RepoRoot{Root: "github.com/shykes/daggerverse.git", Repo: "https://github.com/shykes/daggerverse"},
+				repoRootSubdir: "ci",
+				scheme:         core.SchemeGitSSH,
+				sshusername:    "",
+			},
+		},
+		{
+			urlStr: "git+ssh://github.com/shykes/daggerverse/ci",
+			want: &parsedRefString{
+				modPath:        "github.com/shykes/daggerverse/ci",
+				kind:           core.ModuleSourceKindGit,
+				repoRoot:       &vcs.RepoRoot{Root: "github.com/shykes/daggerverse", Repo: "https://github.com/shykes/daggerverse"},
+				repoRootSubdir: "ci",
+				scheme:         core.SchemeGitSSH,
+				sshusername:    "",
+			},
+		},
+		{
+			urlStr: "ssh://github.com/shykes/daggerverse.git/ci",
```

Comment:
Is there a different between `git+ssh` and `ssh` or are they equivalent?

If they're the same, IMO we should only allow the more explicit `git+ssh`.